src/components/tasks/TaskSearch/TaskSearch.tsx

```diff
@@ -16,84 +23,316 @@ const initialSuggestionCoordinates: SuggestionCoordinates = {
 };
 
 type TaskSearchProps = {
-    onSelect: (tab: Tab) => void;
+    onFilterDropdownSelect: (tab: Tab) => void;
+    filterDropdownActiveTab?: Tab;
     inputValue: string;
-    activeTab?: Tab;
-    onInputChange: (value: string) => void;
     onClickSearchButton: (param?: string) => void;
 };
 
 const TaskSearch = ({
-    onSelect,
+    onFilterDropdownSelect,
+    filterDropdownActiveTab,
     inputValue,
-    activeTab,
-    onInputChange,
     onClickSearchButton,
 }: TaskSearchProps) => {
-    const [modalOpen, setModalOpen] = useState(false);
+    const [filterDropdownModelOpen, setFilterDropdownModelOpen] =
```

Comment:
There are 7 useState's currently in this component. This would be a good candidate to start using useReducer instead to better manage the state.